src/dotnet-aspnet-codegenerator/Program.cs

```diff
@@ -6,127 +6,242 @@
+        ///    2. Try getting the project context for the project (use netcoreapp1.0 as the tfm if not running as dependency command or else use the tfm passed in)
+        ///    3. If not running as dependency command and project context cannot be built using netcoreapp1.0, invoke project dependency command with the first tfm found in the project.json
+        ///    
+        /// Phase 2 ::
+        ///     1. After successfully getting the Project context, invoke the CodeGenCommandExecutor.
+        /// </summary>
+        private static void Execute(string[] args)
+        {
             var app = new CommandLineApplication(false)
             {
                 Name = APPNAME,
                 Description = APP_DESC
             };
+
+
+
+            // Define app Options;
             app.HelpOption("-h|--help");
             var projectPath = app.Option("-p|--project", "Path to project.json", CommandOptionType.SingleValue);
             var packagesPath = app.Option("-n|--nugetPackageDir", "Path to check for Nuget packages", CommandOptionType.SingleValue);
             var appConfiguration = app.Option("-c|--configuration", "Configuration for the project (Possible values: Debug/ Release)", CommandOptionType.SingleValue);
-            
+            var framework = app.Option("-tfm|--targetFramework", "Target Framework to use. (Short folder name of the tfm. eg. net451)", CommandOptionType.SingleValue);
+            var buildBasePath = app.Option("-b|--buildBasePath", "", CommandOptionType.SingleValue);
+            var outputPath = app.Option("-o|--buildOutputPath", "", CommandOptionType.SingleValue);
+            var dependencyCommand = app.Option("-d|--dependencyCommand", "", CommandOptionType.NoValue);
+
             app.OnExecute(() =>
             {
-                var serviceProvider = new ServiceProvider();
-                var context = CreateProjectContext(projectPath.Value());
-                
+
+                #region Phase 1 :: Figure out if we are running as a dependency command (and invoke dependency command if needed)
+
+
                 var configuration = appConfiguration.Value() ?? "Debug";
-                if(configuration != null && !configuration.Equals("Release") && !configuration.Equals("Debug")) 
+                // If its not a dependency command, use netcoreapp1.0 (ignore what is passed in). 
+                NuGetFramework nugetFramework = FrameworkConstants.CommonFrameworks.NetCoreApp10;
+
+                if (isProjectDependencyCommand && !TryGetNugetFramework(framework.Value(), out nugetFramework))
+                {
+                    throw new ArgumentException($"Could not understand the NuGetFramework information. Framework short folder name passed in was: {framework.Value()}");
+                }
+
+                var projectContextInfo = GetProjectContextInformation(projectPath.Value(), nugetFramework);
+
+                if (isProjectDependencyCommand && projectContextInfo.ProjectContext == null)
```

Comment:
The `ProjectContext` is built for a specific TargetFramework, `ProjectContextInformation` gives all available target frameworks. 

Also the intent of having the class was to avoid having to read the project twice (Although I can avoid that by in-lining the code for `GetProjectContextInformation()` method as well)